frontend/src/routes/posts/ShowPost.tsx

```diff
@@ -143,43 +158,54 @@ const ShowPost: React.FC<Props> = ({isReview, addComment, post, setCommentLiked,
         }
     }, [post, setDeleted]);
 
-    const inputChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    const commentInputChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
         setCommentInput(e.target.value);
     }, [setCommentInput]);
 
+    const keyInputChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+        setKeyInput(e.target.value);
+    }, [setKeyInput]);
+
+
     return (
-        <article className="wrapper">
-            <img onDoubleClick={onLike} src={`${BACKEND_API_URL}/${photoURL}`} />
-            <div className="panel">
-                <div className="profile-bar">
-                    <Link to={`/${post.poster.username}`}>
-                        <img src={`${BACKEND_API_URL}/${post.poster.profilePhotoURL}`}></img>
-                    </Link>
-                    <Link to={`/${post.poster.username}`}>
-                        {post.poster.username}
-                    </Link>
-                    {canFollow && <span className="dot-separator">•</span>}
-                    {canFollow && <Button text={post.poster.amFollowing ? 'Unfollow' : 'Follow'} onClick={onFollow} />}
-                </div>
+        <article className="post-wrapper">
+            <div className="post">
+                <img onDoubleClick={onLike} src={`${BACKEND_API_URL}/${photoURL}`} />
+                <div className="panel">
+                    <div className="profile-bar">
+                        <Link to={`/${post.poster.username}`}>
+                            <img src={`${BACKEND_API_URL}/${post.poster.profilePhotoURL}`}></img>
+                        </Link>
+                        <Link to={`/${post.poster.username}`}>
+                            {post.poster.username}
+                        </Link>
+                        {canFollow && <span className="dot-separator">•</span>}
+                        {canFollow && <Button text={post.poster.amFollowing ? 'Unfollow' : 'Follow'} onClick={onFollow} />}
+                    </div>
+
+                    {showComments && <div className="comments">
+                        <ul>
+                        <ShowComment comment={postToComment(post)} key={-1} onLike={onLike} />
+                        {orderComments(post.comments).map(c => <ShowComment comment={c} key={c.id} onReply={onReply} onLike={onCommentLike} />)}
+                        </ul>
+                    </div>}
+
+                    {showComments && <form className="comment-wrapper" onSubmit={onComment}>
+                        <input onChange={commentInputChange} type="text" name="text" className="comment-box" placeholder="Add a comment" ref={commentInputRef} value={commentInput} />
+                        <Button text="Post" disabled={!commentInput} />
+                    </form>}
+
+                    {showComments && <form className="key-wrapper">
```

Comment:
Ovo treba da zavisi od `unlock`, ne od `showComments`. Takođe,
```suggestion
                    {showComments && <form className="key-wrapper" onSubmit={onUnlock}>
```